slider/library.py

```diff
@@ -279,6 +279,29 @@ def lookup_by_md5(self, beatmap_md5):
         """
         return self._read_beatmap(self, beatmap_md5=beatmap_md5)
 
+    def beatmap_from_file(self, path, copy=False):
+        """Returns a beatmap from a file on disk.
+
+        Parameters
+        ----------
+        path : str or pathlib.Path
+            The path to the file to create the beatmap from.
+        copy : bool
+            Should the file be copied to the library's beatmap directory?
+
+        Returns
+        -------
+        beatmap : Beatmap
+            The beatmap represented by the given file.
+        """
+        beatmap = Beatmap.from_path(path)
+
+        if copy:
+            data = open(path, 'rb').read()
```

Comment:
currently this opens and reads the file twice. We can always open and read the file manually and then use `Beatmap.parse` to convert the raw data into a `Beeatmap` object (https://github.com/llllllllll/slider/blob/master/slider/beatmap.py#L1860).